onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -231,12 +359,158 @@ static Status InlineNodes(Graph& graph, bool& modified_graph) {
+
+    Node& fused_node = graph.BeginFuseSubGraph(indexed_sub_graph, node_name);
+    fused_node.SetExecutionProviderType(type);
+
+    // create filtered graph viewer for this set of nodes
+    //
+    // TODO: Could avoid the topological sort in the GraphViewer ctor by constructing from an existing
+    // GraphViewer instance instead of the Graph (copying the topological order instead of recalculating).
+    viewers.push_back(onnxruntime::make_unique<GraphViewer>(graph, indexed_sub_graph));
+    nodes_and_viewers.push_back(IExecutionProvider::FusedNodeAndGraph{fused_node, *viewers.back()});
+  }
+
+  std::vector<NodeComputeInfo> node_compute_funcs;
+  node_compute_funcs.reserve(nodes_and_viewers.size());
+
+  ORT_RETURN_IF_ERROR(current_ep.Compile(nodes_and_viewers, node_compute_funcs));
+
+  if (node_compute_funcs.size() != nodes_and_viewers.size()) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, type, " did not return correct number of compiled functions");
+  }
+
+  for (size_t j = 0, end = nodes_and_viewers.size(); j < end; j++) {
+    Node& node = nodes_and_viewers[j].fused_node;
+
+    ORT_RETURN_IF_ERROR(func_mgr.AddFuncInfo(node.Name(), std::move(node_compute_funcs[j])));
+
+    const auto& cur_capability = capabilities[j];
+    const IndexedSubGraph& indexed_sub_graph = *cur_capability->sub_graph;
+    const IndexedSubGraph::MetaDef& metadef = *indexed_sub_graph.GetMetaDef();
+
+    KernelDefBuilder builder;
+    BuildFusedKernelDef(builder, metadef, type);
+    auto kernel_def = builder.Build();
+
+    // save hash so SessionState can find the kernel. each kernel name should be unique
+    if (compiled_kernel_hashes.insert({metadef.name, kernel_def->GetHash()}).second == false) {
+      ORT_THROW("Existing entry in compiled kernel hashes for ", metadef.name,
+                ". Execution Provider must generate unique names across the entire model.");
+    }
+
+    ORT_RETURN_IF_ERROR(fused_kernel_registry.Register(
+        KernelCreateInfo(std::move(kernel_def), static_cast<KernelCreatePtrFn>(
+                                                    [](const OpKernelInfo& info) -> OpKernel* {
+                                                      return new FunctionKernel(info);
+                                                    }))));
+
+    // now that we're done compiling we can remove the original nodes from the Graph and wire in the new one
+    graph.FinalizeFuseSubGraph(indexed_sub_graph, node);
+  }
+
+  return Status::OK();
+}
+
+// Simplified partitioning where custom EPs may produce compiled nodes.
+// EPs with static kernels do not need to be processed as their kernels are matched via hash information serialized
+// as part of the ORT format model.
+Status GraphPartitioner::PartitionOrtFormatModel(
+    Graph& graph, FuncManager& func_mgr,
+    KernelRegistry& fused_kernel_registry,
+    std::unordered_map<std::string, uint64_t>& compiled_kernel_hashes) const {
+  int unique_id = 0;
+
```

Comment:
Move this into GraphPartitioner::Partition? 
And rename to fused_node_id to make it clear?
Maybe some comments about this is unique per-model?